sa4u_z3/main.py

```diff
@@ -272,6 +272,7 @@ def main():
     )
 
     signal.signal(signal.SIGHUP, HUP_signal_handler)
+    signal.signal(signal.SIGTERM, TERM_signal_handler)
```

Comment:
Why aren't we able to use `signal.SIG_DFL`?

Reply:
After running tests, having our own code to handle SIGTERM is significantly faster than the default handler.

Reply:
Huh. I wonder why. Perhaps the default handler waits to exit until after something important happens? Either way, we can disregard this for now.